app/qml/FeatureForm.qml

```diff
@@ -441,16 +446,39 @@ Item {
         }
 
         Label {
-          id: constraintDescriptionLabel
+          id: fieldHelperText
+
+          property string constraintText: {
+            if ( ValueValidity === FormItem.ValueOutOfRange )
+              return qsTr( 'Number is outside of specified range' )
+            else if ( ValueValidity === FormItem.InvalidValue )
+              return qsTr( 'Value is not valid' )
+
+            if ( ConstraintDescription )
+              return ConstraintDescription
+
+            return ''
+          }
+
+          property bool shouldShowConstraintText: {
+            if ( ( !ConstraintHardValid || !ConstraintSoftValid ) && !!ConstraintDescription )
+              return true
+
+            if ( ValueValidity !== FormItem.ValidValue )
+              return true
+
+            return false
+          }
```

Comment:
logic should be in c++; also misleading name of property

Reply:
Will be in #1507